src/jshint.js

```diff
@@ -1419,12 +1425,23 @@ var JSHINT = (function() {
     var x = symbol(s, 150);
 
     x.led = function(left) {
+      // this = suffix e.g. "++" punctuator
+      // left = symbol operated e.g. "a" identifier or "a.b" punctuator
+
       if (state.option.plusplus) {
         warning("W016", this, this.id);
       } else if ((!left.identifier || isReserved(left)) && left.id !== "." && left.id !== "[") {
         warning("W017", this);
       }
 
+      // detect modification of a const.
+      // in the case of a.b, left will be the "." punctuator
```

Comment:
> This might be shocking, but I have been active on github for a number of years

yep, me too

> I'm aware of this.

you never know! sorry, I didn't mean to offend.



> $('.discussion-item-toggle-open').click() didn't find your comment

good tip, bookmarklet added.



> Anyways, you've been asked to add some tests and comment your stuff clearly, I think that's the only thing blocking landing this

yep both those things are already done